Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts

```diff
@@ -0,0 +1,135 @@
+        });
+    }
+
+    public getArtifactItems(artifactItem: models.ArtifactItem): Promise<models.ArtifactItem[]> {
+        return new Promise(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = await this._getItems(this._container, artifactItem.path);
+            resolve(items);
+        });
+    }
+
+    public getArtifactItem(artifactItem: models.ArtifactItem): Promise<stream.Readable> {
+        return new Promise((resolve, reject) => {
+            var readStream: stream.Readable = this._blobSvc.createReadStream(this._container, artifactItem.path, null);
+            resolve(readStream);
+        });
+    }
+
+    private _ensureContainerExistence(): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            if (!this._isContainerExists) {
+                var self = this;
+                this._blobSvc.createContainerIfNotExists(this._container, function (error, result, response) {
+                    if (!!error) {
+                        console.log(tl.loc("FailedToCreateContainer", self._container, error.message));
+                        reject(error);
+                    } else {
+                        self._isContainerExists = true;
+                        console.log(tl.loc("CreatedContainer", self._container));
+                        resolve();
+                    }
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    private _getItems(container: string, parentRelativePath?: string): Promise<models.ArtifactItem[]> {
+        var promise = new Promise<models.ArtifactItem[]>((resolve, reject) => {
+            var items: models.ArtifactItem[] = [];
+
+            this._blobSvc.listBlobsSegmentedWithPrefix(container, parentRelativePath, null, (error, result) => {
+                if (!!error) {
+                    console.log(tl.loc("FailedToListItemInsideContainer", container, error.message));
+                    reject(error);
+                } else {
+                    console.log(tl.loc("SuccessFullyFetchedItemList"));
+                    if(result.continuationToken){
+                        tl.warning(tl.loc("ArtifactItemsTruncationWarning"));
+                    }
+                    items = this._convertBlobResultToArtifactItem(result.entries);
+                    resolve(items);
+                }
+            });
+        });
+
+        return promise;
+    }
+
+    private _convertBlobResultToArtifactItem(blobResult: azureStorage.BlobService.BlobResult[]): models.ArtifactItem[] {
+        var artifactItems: models.ArtifactItem[] = new Array<models.ArtifactItem>();
+        blobResult.forEach(element => {
+            var artifactitem: models.ArtifactItem = new models.ArtifactItem();
+            artifactitem.itemType = models.ItemType.File;
+            artifactitem.fileLength = parseInt(element.contentLength);
+            artifactitem.lastModified = new Date(element.lastModified + 'Z');
```

Comment:
is this date UTC?

Reply:
yes, it gets converted to utc when we add "z" in the end